torch/nn/parallel/distributed.py

```diff
@@ -464,6 +473,11 @@ def __init__(self, module, device_ids=None,
         if self.device_ids is not None and len(self.device_ids) > 1:
             dist._verify_replicas_within_process(parameters, expect_sparse_gradient)
         dist._verify_model_across_ranks(self.process_group, parameters)
+        if _GLOO_AVAILABLE and dist._get_debug_mode() != dist._DistributedDebugLevel.OFF:
+            # Run monitored barrier in debug mode. This will ensure all ranks
+            # have same model shapes before continuing with init/training.
+            dist.monitored_barrier(group=self._cpu_pg, timeout=30)
```

Comment:
nit: Could we add something like LOG("calling  monitored_barrier after _verify_model_across_ranks....") before dist.monitored_barrier() call?